### PR TITLE
Hotfix for missing session recording events

### PR DESCRIPTION
**Pull request by macobo**
Temporarily solves https://github.com/PostHog/posthog/issues/2926,
though the error will rear it's head again with plugin-based ingestion.

Opened https://github.com/PostHog/posthog/issues/3068 for long-term
solution.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by macobo** [comment]
Ping @fuziontech, let me know if any context needed here?

**Comment by fuziontech** [comment]
Copying our conversation here for posterity!

tl;dr - I have an aversion to kwargs because they leak the abstraction we are trying to make here. Right now it would be kind of easy to swap out clickhouse-driver if we needed to or the connection pooler library because there is a clear wiring between the client's interface and the underlying lib. If you add kwargs to the client call `sync_execute` users can add whatever params they want that clickhouse-driver accepts. If we ever want to swap out the driver lib we will need to go spelunking for kwargs users are passing to the client lib which is not ideal.

Thinking on where we should set the setting here I think it should be our clickhouse default to allow larger payloads to be inserted. We should probably just bump this setting as a default on clickhouse, especially if the long term goal is to put these kinds of messages on kafka since kafka payloads end up locally on clickhouse as batched inserts.

**Comment by fuziontech** [comment]
only other question I have here is are you sure `max_query_size` is the right setting?

https://clickhouse.tech/docs/en/operations/settings/settings/#settings-max_query_size

> The maximum part of a query that can be taken to RAM for parsing with the SQL parser.
> The INSERT query also contains data for INSERT that is processed by a separate stream parser (that consumes O(1) RAM), which is not included in this restriction.
> 
> Default value: 256 KiB.

Which sounds like inserts are not constrained by this restriction?

**Comment by macobo** [comment]
> only other question I have here is are you sure max_query_size is the right setting?

It is. I followed repro steps from https://github.com/PostHog/posthog/issues/2927#issuecomment-763578622 on a domain which consistently produced >1MB full snapshots. Previously they were emitted but never read, this fixes that issue.

**Comment by macobo** [comment]
I've updated the code to use an explicit settings arg - as mentioned in slack, disagree but not enough to warrant a longer back-and-forth.

Given https://github.com/PostHog/posthog/issues/3068 I think this should be fine for now and I'd love to avoid iterating/spending time on this. We have a temporary working solution for a problem we're hoping to solve soon elsehow anyways. 

A better fix is blocked currently by 1) plugin-based ingestion and 2) getting kafka over to aws, I'm participating in neither project currently. The alternative you've suggested (bumping setting on clickhouse + locally) also requires work from you - but you're a constrained resource and splitting focus like this would not solve anything. As you said - we need to fix it anyways later, so why split your focus now?

As Voltaire said - Perfect is the enemy of the good. Let's ship this, clients are getting a bad experience!
